lib/classifier-reborn/bayes.rb

```diff
@@ -19,24 +21,27 @@ class Bayes
     #   threshold:        0.0    Default threshold, only used when enabled
     #   enable_stemmer:   true   When false, disables word stemming
     def initialize(*args)
-      @categories = {}
+      initial_categories = []
       options = { language:         'en',
                   auto_categorize:  false,
                   enable_threshold: false,
                   threshold:        0.0,
-                  enable_stemmer:   true
+                  enable_stemmer:   true,
+                  redis_con_str:    ''
```

Comment:
I think the best option here is to use dependency injection. Thus the user does:

```ruby
memory_backed_classifier = ClassifierReborn::Bayes.new
redis_backed_classifier = ClassifierReborn::Bayes.new(:backend => ClassifierReborn::BayesRedisBackend.new("localhost", "6987"))
```

So they pass in a `:backend` key which is an instance of the relevant backend they wish to use. Then they can configure Redis and we don't have to worry about any backend configurations in this class.